engines/scumm/detection.cpp

```diff
@@ -252,6 +253,53 @@ static const ExtraGuiOption enableOriginalGUI = {
 	0
 };
 
+Common::String ScummMetaEngineDetection::parseAndCustomizeGuiOptions(const Common::String &optionsString, const Common::String &domain) const {
+	Common::String result = MetaEngineDetection::parseAndCustomizeGuiOptions(optionsString, domain);
+	const char *defaultRenderOption = GUIO_RENDERVGA;
+
+	const Common::Platform platform = Common::parsePlatform(ConfMan.get("platform", domain));
+	const Common::String extra = ConfMan.get("extra", domain);
+
+	// Add default rendermode option for target. We don't put the default mode into the
+	// detection tables, due to the amount of targets we have. It it more convenient to
+	// add the option here.
+	switch (platform) {
+	case Common::kPlatformAmiga:
+		defaultRenderOption = GUIO_RENDERAMIGA;
+		break;
+	case Common::kPlatformApple2GS:
+		defaultRenderOption = GUIO_RENDERAPPLE2GS;
+		break;
+	case Common::kPlatformMacintosh:
+		defaultRenderOption = GUIO_RENDERMACINTOSH;
+		break;
+	case Common::kPlatformFMTowns:
+		defaultRenderOption = GUIO_RENDERFMTOWNS;
+		break;
+	case Common::kPlatformAtariST:
+		defaultRenderOption = GUIO_RENDERATARIST;
+		break;
+	case Common::kPlatformDOS:
+		if (extra.equalsIgnoreCase("EGA") || extra.equalsIgnoreCase("V1") || extra.equalsIgnoreCase("V2"))
+			defaultRenderOption = GUIO_RENDEREGA;
+		break;
+	case Common::kPlatformUnknown:
+		// For targets that don't specify the platform we stick with default VGA.
+		break;
+	default:
+		// Clear this for platforms that don't have a specific render option (SegaCD, NES, ...)
+		defaultRenderOption = nullptr;
```

Comment:
IMHO it's slightly more readable to initialize to nullptr initially, and then let the default here set VGA.

Reply:
Sure, I have changed it...